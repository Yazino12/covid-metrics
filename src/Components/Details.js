import React from 'react';
import { NavLink } from 'react-router-dom';
import wallpaper from '../assets/covid-1.jpg';

const Details = () => (
  <div className="details">
    <div className="header-details">
      <img src={wallpaper} alt="Covid-19 img" />
      <div className="country-cases">
        <h2>Somalia</h2>
        <p>
          12,457
          <span>cases</span>
        </p>
      </div>
    </div>
    <div className="stats-by-country">
      <p className="p">STATS BY COUNTRY</p>
      <div className="stats">
        <div className="stat population">
          <h4>population</h4>
          <p>35530081</p>
        </div>
        <div className="stat confirmed">
          <h4>confirmed</h4>
          <p>35530081</p>
        </div>
        <div className="stat recovered">
          <h4>recovered</h4>
          <p>35530081</p>
        </div>
        <div className="stat deaths">
          <h4>deaths</h4>
          <p>35530081</p>
        </div>
      </div>
    </div>
    <div className="links">
      <NavLink to="/" className="link">
        Home
      </NavLink>
    </div>
  </div>
);

export default Details;
