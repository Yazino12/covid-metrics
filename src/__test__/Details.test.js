import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Details from '../Components/Details';
import store from '../Redux/configStore';

describe('Details tests', () => {
  test('should render', () => {
    const details = render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
        ,
      </Provider>
    );

    expect(details).toMatchSnapshot();
  });

  test('Checks if details content is there by checking text', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
        ,
      </Provider>
    );

    expect(
      screen.getByRole('img', { name: /Covid-19 img/i })
    ).toBeInTheDocument();
  });
});
