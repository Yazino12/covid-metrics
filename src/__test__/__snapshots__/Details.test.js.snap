// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <svg
          class="navigateBack"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
          />
        </svg>
        <div
          class="right"
        >
          <svg
            class="voiceIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
            />
          </svg>
          <svg
            class="settings"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="3"
            />
            <path
              d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
            />
          </svg>
        </div>
      </header>
      <div
        class="details"
      >
        <div
          class="header-details"
        >
          <img
            alt="Covid-19 img"
            src="covid-1.jpg"
          />
          <div
            class="country-cases"
          >
            <h2 />
            <p>
              <span>
                cases
              </span>
            </p>
          </div>
        </div>
        <div
          class="stats-by-country"
        >
          <p
            class="p"
          >
            STATS BY COUNTRY
          </p>
          <div
            class="stats"
          >
            <div
              class="stat confirmed"
            >
              <h4>
                confirmed
              </h4>
              <p />
            </div>
            <div
              class="stat recovered odds"
            >
              <h4>
                recovered
              </h4>
              <p />
            </div>
            <div
              class="stat deaths"
            >
              <h4>
                deaths
              </h4>
              <p />
            </div>
            <div
              class="stat population odds"
            >
              <h4>
                new confirmed
              </h4>
              <p />
            </div>
            <div
              class="stat population odds"
            >
              <h4>
                new confirmed
              </h4>
              <p />
            </div>
            <div
              class="stat population odds"
            >
              <h4>
                new confirmed
              </h4>
              <p />
            </div>
            <div
              class="stat population"
            >
              <h4>
                updated
              </h4>
              <p />
            </div>
          </div>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <header>
      <svg
        class="navigateBack"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
        />
      </svg>
      <div
        class="right"
      >
        <svg
          class="voiceIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
          />
        </svg>
        <svg
          class="settings"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </div>
    </header>
    <div
      class="details"
    >
      <div
        class="header-details"
      >
        <img
          alt="Covid-19 img"
          src="covid-1.jpg"
        />
        <div
          class="country-cases"
        >
          <h2 />
          <p>
            <span>
              cases
            </span>
          </p>
        </div>
      </div>
      <div
        class="stats-by-country"
      >
        <p
          class="p"
        >
          STATS BY COUNTRY
        </p>
        <div
          class="stats"
        >
          <div
            class="stat confirmed"
          >
            <h4>
              confirmed
            </h4>
            <p />
          </div>
          <div
            class="stat recovered odds"
          >
            <h4>
              recovered
            </h4>
            <p />
          </div>
          <div
            class="stat deaths"
          >
            <h4>
              deaths
            </h4>
            <p />
          </div>
          <div
            class="stat population odds"
          >
            <h4>
              new confirmed
            </h4>
            <p />
          </div>
          <div
            class="stat population odds"
          >
            <h4>
              new confirmed
            </h4>
            <p />
          </div>
          <div
            class="stat population odds"
          >
            <h4>
              new confirmed
            </h4>
            <p />
          </div>
          <div
            class="stat population"
          >
            <h4>
              updated
            </h4>
            <p />
          </div>
        </div>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
